packages/langium/src/lsp/completion/follow-element-computation.ts

```diff
@@ -4,25 +4,34 @@
  * terms of the MIT License, which is available in the project root.
  ******************************************************************************/
 
+import { IToken } from 'chevrotain';
 import * as ast from '../../grammar/generated/ast';
-import { Cardinality, isArray, isOptional } from '../../grammar/grammar-util';
+import { Cardinality, getCrossReferenceTerminal, isArray, isOptional, terminalRegex } from '../../grammar/grammar-util';
 
 /**
  * Calculates any features that can follow the given feature stack.
  * This also includes features following optional features and features from previously called rules that could follow the last feature.
  * @param featureStack A stack of features starting at the entry rule and ending at the feature of the current cursor position.
  * @returns Any `AbstractElement` that could be following the given feature stack.
  */
-export function findNextFeatures(featureStack: ast.AbstractElement[]): ast.AbstractElement[] {
-    return findNextFeaturesInternal(featureStack, new Map<ast.AbstractElement, Cardinality>());
+export function findNextFeatures(featureStack: ast.AbstractElement[], unparsedTokens: IToken[]): ast.AbstractElement[] {
+    const context: InterpretationContext = {
+        stacks: [featureStack],
+        tokens: unparsedTokens
+    };
+    interpreteTokens(context);
+    const nextStacks = findNextFeatureStacks(context.stacks);
+    // We only need the last element of each stack
+    return nextStacks.map(e => e[e.length - 1]);
```

Comment:
Personal preference, `e => e.pop()`

Reply:
I feel like `pop` implies that we are interested in the other part of the array as well (due to its side effects). However, we only want the last element, so I'll keep it like this. `pop` is also slower.